### DISCUSSION TITLE
More FPS = More Speed (game speeds up when V-Sync is off)

### DISCUSSION THREAD

**Discussion by thebashforkbomber** [FakeFishGames/Barotrauma#13629] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

The title says it all. The higher the FPS, the faster the game plays.
This is probably because of overperformance.

### Reproduction steps

1. Open the game
2. Open settings
3. Disable V-Sync.
4. (optional) Set up a save and observe that your character moves faster, and shoots too.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Other

### -

LITERALLY EVERYWHERE WHAT THE

### Version

v1.2.8.0 (Winter Update hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
No errors produced.
```

**Comment by thebashforkbomber** [comment]
NOTE: The only mods I had is CL Lua and Performance fix with Immersive Ammunition boxes. Nothing increases player speed.

**Comment by Regalis11** [comment]
Thank you for the report!

This shouldn't be the case, the physics run at a fixed rate (60 updates per second) that's not tied to the framerate in any way. I also haven't been able to reproduce this: the game seems to run at the same speed regardless of the framerate.

I would maybe suspect this is caused by Performance fix - it's known to do all sorts of weird things with the game's code and break a lot of things.

**Comment by thebashforkbomber** [comment]
> I would maybe suspect this is caused by Performance fix - it's known to do all sorts of weird things with the game's code and break a lot of things.

Performance fix isn't the problem; with V-Sync on, it returns to normal (normal game speed)
[Here's a vid that I managed to make.
](https://dl.dropbox.com/scl/fi/7puc6knv636i10s55yhdi/what-the.mp4?rlkey=hgl5txg09mo94te06z9nlluj6&dl=1)
BTW, the video isn't sped up in any way. Its real time footage.

**Comment by Regalis11** [comment]
Have you tried disabling Performance fix and checking if the game still does that? I would be willing to bet that it's caused by the mod, because many players and us devs play with V-Sync off and have never run into an issue like this. Performance fix does all kinds of strange things with how the game handles timing, and I wouldn't be surprised if it was behind this issue too.

**Comment by Regalis11** [comment]
Here's how the _vanilla game_ runs at my end with V-Sync disabled and the FPS well above 200:
https://github.com/FakeFishGames/Barotrauma/assets/14351070/c77963ab-5438-4ae1-8889-582b690612ab

**Comment by thebashforkbomber** [comment]
[Here, no performance fix, V-Sync disabled](https://www.dropbox.com/scl/fi/l6d0w5jb30l6yraw7hcex/Faster.mp4?rlkey=3vsyj39m5j61qhb400i59irpw&dl=0)

**Comment by Regalis11** [comment]
Hmm, that's very strange. Are you completely sure it's entirely uninstalled? To my understanding, uninstalling lua mods entirely is a bit involved, and just disabling them through the "Mods" tab won't work. Could you take a screenshot of the main menu to show what exact build of the game you are running?

**Comment by thebashforkbomber** [comment]
![изображение](https://github.com/FakeFishGames/Barotrauma/assets/102991955/a07b7b34-840b-4ad7-80a0-0c37269204ea)

**Comment by Regalis11** [comment]
Thank you! That seems to be the latest, up-to-date vanilla build. :/ I can't think of any other possible explanations for this off the top of my head, but since this doesn't seem to be a widespread issue or happen on all systems, I suppose it could have something to do with your system or perhaps some 3rd party software interfering with the game in some way. I'm sorry I can't give you any more assistance on this for the time being, but we'll keep the ticket open in case we'll be able to find some more clues for diagnosing this. Sorry for the inconvenience!
